Core module updated

obedjamir opened the pull request:
Database entity classes created
Repository interfaces created

obedjamir committed:
Core module updated
Database entity classes created
Repository interfaces created

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/Trip.java:
```diff
@@ -0,0 +1,45 @@
+    private Long tripId;
+
+    @Column(name="trip_name")
+    @NotBlank(message = "tripName cannot be blank")
```
Dont use @NotBlank here. Column has an attribute nullable set that.

obedjamir replied:
This has been resolved

obedjamir committed:
Core module updated
@LastmodifiedDate and @CreatedDate implemented
Model classes implements serializable

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java:
```diff
@@ -0,0 +1,52 @@
+    private String location;
+
+    @Column(name="activity_time",nullable = false)
+    private Time activityTime;
```
Make this Date

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/TripUserMapping.java:
```diff
@@ -0,0 +1,47 @@
+    private Long userId;
+
+    @Column(name="user_role",nullable = false)
+    private String userRole;
```
Make this an enum

obedjamir committed:
Core module updated
userRole is now an enum
activityTime is now Date type

PinakiGhosh commented on core/src/main/java/com/bankbazaar/tripmanager/model/TripActivity.java:
```diff
@@ -30,7 +30,7 @@ public class TripActivity implements Serializable {
 
     @Column(name="activity_time",nullable = false)
-    private Time activityTime;
+    private Data activityTime;
```
Date?

obedjamir replied:
Oh my bad. missed that one

obedjamir committed:
Typo correction
Changed Data to Date
